blockchain/processing.py

```diff
@@ -255,6 +299,8 @@ def _execute_phase_1(self, config, current_block_id):
             # update status of approved transactions
             for tx in approved_transactions:
                 tx["header"]["status"] = "approved"
+                if tx["header"]["transaction_type"] == "TT_SUB_REQ":
```

Comment:
I'm thinking that we'll want to replace the `approve_block` and `approve_transaction` conditionals above this section - 
In their stead, put a loop through all transactions and call functions registered in the `smart_contracts` dictionary per their respective transaction types.

We'll then want to pull the config at initialization and place/overwrite the `smart_contracts` dict with config values.

(when we later allow a reserved trans type to use new smart contracts, this will still just work).